concent_api/signing_service_integration_test_recovery.py

```diff
@@ -0,0 +1,242 @@
+
+        # Create GolemMessageFrame with correct TransactionSigningRequest.
+        golem_message_frame = create_golem_message_frame_with_correct_transaction_signing_request(
+            request_id=get_current_utc_timestamp(),
+        )
+
+        # Send message through wrapper and receive deserialized response.
+        response = send_message_to_middleman_and_receive_response(
+            message=golem_message_frame,
+            config=config,
+            concent_private_key=CONCENT_PRIVATE_KEY,
+            concent_public_key=CONCENT_PUBLIC_KEY,
+        )
+
+        # Check response.
+        assert_condition(type(response), GolemMessageFrame, f'Deserialized response type is {type(response)} instead of GolemMessageFrame.')
+        assert_condition(type(response.payload), SignedTransaction, f'Deserialized response payload type is {type(response.payload)} instead of SignedTransaction.')
+        assert_condition(response.request_id, golem_message_frame.request_id, f'Deserialized response request_id is {response.request_id} instead of {golem_message_frame.request_id}.')
+
+    finally:
+        middleman_process.kill()
+        signing_service_process.kill()
+
+
+@count_fails
+def test_case_1_middleman_recovery(config: ConfigParser, **kwargs: Any) -> None:
+    """
+    1. Spawn MiddleMan and SigningService processes.
+    2. Client sends GolemMessageFrame with correct TransactionSigningRequest to MiddleMan.
+    3. Middleman is restarted. The connection and latest message is lost.
+    4. Client sends GolemMessageFrame with correct TransactionSigningRequest to MiddleMan.
+    5. Client receives response for latest message.
+    """
+
+    try:
+        middleman_process = run_middleman()
+        signing_service_process = run_signing_service()
+
+        # Waiting for MiddleMan and SigningService to start.
+        sleep(SLEEP_TIME_AFTER_SPAWNING_OR_KILLING_PROCESS)
+
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
+
+        # Create GolemMessageFrame with correct TransactionSigningRequest.
+        golem_message_frame = create_golem_message_frame_with_correct_transaction_signing_request(
+            request_id=get_current_utc_timestamp(),
+        )
+
+        client_socket.connect(
+            (
+                config.get(Components.MIDDLEMAN.value, 'host'),
+                int(config.get(Components.MIDDLEMAN.value, 'port')),
+            )
+        )
+        send_over_stream(
+            connection=client_socket,
+            raw_message=golem_message_frame,
+            private_key=CONCENT_PRIVATE_KEY,
+        )
+
+        middleman_process.kill()
+
+        # Waiting for MiddleMan to finish.
+        sleep(SLEEP_TIME_AFTER_SPAWNING_OR_KILLING_PROCESS)
+
+        middleman_process = run_middleman()
+
+        # Waiting for MiddleMan to start.
+        sleep(SLEEP_TIME_AFTER_SPAWNING_OR_KILLING_PROCESS)
```

Comment:
`SLEEP_TIME_AFTER_SPAWNING_OR_KILLING_PROCESS` is 10, so this test takes at least 30 seconds. Is it possible to do it faster?

Reply:
10 is safe value, because this test might be run on different machines. For example when I was running it on debug mode in Pycharm, which always make it slower, 10 seconds was not enough.

Reply:
ok, I didn't meant to decrease this time, only thought about a return message from app or check if process is started or killed. But you answered it above